conans/client/installer.py

```diff
@@ -90,13 +89,12 @@ def prepare_build(self):
             mkdir(self.build_folder)
             self._conan_file.source_folder = self.source_folder
         else:
-            if platform.system() == "Windows" and os.getenv("CONAN_USER_HOME_SHORT") != "None":
-                from conans.util.windows import ignore_long_path_files
-                ignore = ignore_long_path_files(self.source_folder, self.build_folder, self._out)
-            else:
-                ignore = None
-
-            shutil.copytree(self.source_folder, self.build_folder, symlinks=True, ignore=ignore)
+            try:
+                shutil.copytree(self.source_folder, self.build_folder, symlinks=True)
+            except Exception as e:
+                msg = str(e)
+                msg += "\nConsider using short_paths=True if paths too long" if "206" in msg else ""
```

Comment:
The `206` looks like some kind of magic number, I would add a comment with a link or maybe check for `WindowsError: [Error 206]` to make it clearer (maybe the full error message).

Also, can the message be assertive? if the Windows error is `WindowsError: [Error 206] The filename or extension is too long` probably we can tell the user without doubts that **the error is because of long paths** and recommend the `short_paths` solution.

Reply:
Fun fact it is not a ``WindowsError``. It is a ``shutil.Error`` wrapping a system error, with no further metadata, because it comes from the system. The only thing that seems to be useful is the system error number, cause I guess the error string might be localized.